spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,201 @@
+
+            # check save connection settings
+            dlg.save_layout.setChecked(save_settings)
+
+            # Press ok and save connection settings
+            qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                             Qt.LeftButton)
+
+            return dlg
+
+    def teardown():
+        """Clear existing-kernel config and keyring passwords"""
+        CONF.remove_section("existing-kernel")
+
+        try:
+            import keyring
+            keyring.set_password("existing_kernel",
+                                 "ssh_key_passphrase", "")
+            keyring.set_password("existing_kernel",
+                                 "ssh_password", "")
+        except Exception:
+            pass
+
+    request.addfinalizer(teardown)
+    return DialogFactory()
+
+
+cf_path = "test_cf_path"
+un = "test_username"
+hn = "test_hostname"
+pn = 123
+kf = "test_kf"
+kfp = "test_kfp"
+pw = "test_pw"
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+def test_kernel_connection_dialog_remember_input_ssh_passphrase(
+        qtbot, connection_dialog_factory):
+    """Test that the dialog remember user's kernel connection
+       settings and ssh key passphrase when the user checks the
+       save checkbox."""
+
+    connection_dialog_factory.submit_filled_dialog(use_keyfile=True,
+                                                 save_settings=True)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get_default_dialog()
+    assert new_dlg.cf.text() == cf_path
+    assert new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == hn
+    assert new_dlg.un.text() == un
+    assert new_dlg.pn.text() == str(pn)
+    assert new_dlg.kf.text() == kf
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == kfp
+
+
+def test_kernel_connection_dialog_doesnt_remember_input_ssh_passphrase(
+        qtbot, connection_dialog_factory):
+    """Test that the dialog doesn't remember user's kernel
+       connection settings and ssh key passphrase when the user doesn't
+       check the save checkbox."""
+
+    connection_dialog_factory.submit_filled_dialog(use_keyfile=True,
+                                                 save_settings=False)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get_default_dialog()
+    assert new_dlg.cf.text() == ""
+    assert not new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == ""
+    assert new_dlg.un.text() == ""
+    assert new_dlg.pn.text() == "22"
+    assert new_dlg.kf.text() == ""
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == ""
+
+
+def test_kernel_connection_dialog_remember_input_password(
```

Comment:
*remembers, remove at least ``input``

Reply:
test_kernel_connection_dialog_remember_password is misleading since it remembers more than just password, but this is the case where it also remembers password (instead of ssh keyphrase)

Reply:
Okay, so what would you suggest?